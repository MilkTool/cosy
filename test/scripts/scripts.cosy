
struct Point {
	mut x Num = 2
    var y = 4
    var a Array Num
}

struct Fake {
	mut x Num = 2
    mut a = '4'
}

{
    mut point = Point { x = 5, y = 13, a = [] } // ok
    print point
}

{
    mut point = Point { x = 5, y = 13, x = 3 } // fails
    print point
}

// {
//     mut point = Point { x = 5, y = 13 } // TODO: Should fail
//     print point
// }

{
    mut point = Point {} // ok
    point.x = 555
    point.y = 7 // TODO: Should fail because y is not mut
    point.z = 8 // fails
    point.w = fn() { // fails
    	print 'hello'
    }
    print point
    print point.x + ', ' + point.y
}

{
    var p = Point {}
    p = Fake {} // fails
}

{
    fn struct_as_arg(s) {
        print s
    }
    struct_as_arg(Point { x = 3, y = 4 })
}

{
    struct Test {
        var a // TODO: Should be var a Point
        var b Str
    }

    print Test { a = Point { x = 666, y = 777 }, b = 'hej' }
}
