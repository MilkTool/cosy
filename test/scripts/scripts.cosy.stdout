[line 19] Error at "x": Member already assigned in initializer.
[line 32] Error at "z": No member named "z" in struct of type Struct { y Num, x Num, a Array(Num) }
[line 33] Error at "w": No member named "w" in struct of type Struct { y Num, x Num, a Array(Num) }
[line 42] Error at "p": Cannot assign Struct { x Num, a Str } to Struct { y Num, x Num, a Array(Num) }
[line 59] Error at "a": Expected value to be of type Unknown but got Struct { y Num, x Num, a Array(Num) }
